test/jdk/sun/security/ec/ECDHKeyAgreementParamValidation.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (C) 2024, THL A29 Limited, a Tencent company. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8320449
+ * @summary ECDHKeyAgreement should validate parameters before assigning them to fields.
+ * @run junit ECDHKeyAgreementParamValidation
```

Comment:
Most security regression tests don't use junit. I think it would be better to not rely on it. There is a similar asserts library for tests that you can use in `test/lib/jdk/test/lib/Asserts.java`.

Reply:
I originally didn't depend on JUnit. But this tool can easily execute multiple test cases independently.
A single failed case doesn't make the whole test fail fast, and all cases always be executed.

Reply:
Fair point, although there are ways to workaround that w/o junit.

AFAICT, this will be the first security test to depend on junit. @rhalade are you ok with this?

Reply:
Just updated this test and not used JUnit.